packages/neos-ui-guest-frame/src/initializeGuestFrame.js

```diff
@@ -0,0 +1,108 @@
+    const inlineEditorRegistry = globalRegistry.get('inlineEditors');
+    const guestFrameWindow = getGuestFrameWindow();
+    const documentInformation = Object.assign({}, guestFrameWindow['@Neos.Neos.Ui:DocumentInformation']);
+    const nodes = Object.assign({}, guestFrameWindow['@Neos.Neos.Ui:Nodes'], {
+        [documentInformation.metaData.contextPath]: documentInformation.metaData.documentNodeSerialization
+    });
+
+    yield put(actions.CR.Nodes.add(nodes));
+
+    yield put(actions.UI.ContentCanvas.setContextPath(documentInformation.metaData.contextPath, documentInformation.metaData.siteNode));
+    yield put(actions.UI.ContentCanvas.setPreviewUrl(documentInformation.metaData.previewUrl));
+    yield put(actions.CR.ContentDimensions.setActive(documentInformation.metaData.contentDimensions.active));
+
+    getGuestFrameBody().addEventListener('click', e => {
+        const clickPath = Array.prototype.slice.call(e.path);
+        const isInsideInlineUi = clickPath.some(domNode =>
+            domNode &&
+            domNode.getAttribute &&
+            domNode.getAttribute('data-__neos__inlineUI')
+        );
+        const selectedDomNode = clickPath.find(domNode =>
+            domNode &&
+            domNode.getAttribute &&
+            domNode.getAttribute('data-__neos-node-contextpath')
+        );
+
+        if (isInsideInlineUi) {
+            // Do nothing, everything OK!
+        } else if (selectedDomNode) {
+            const contextPath = selectedDomNode.getAttribute('data-__neos-node-contextpath');
+            const fusionPath = selectedDomNode.getAttribute('data-__neos-fusion-path');
+
+            store.dispatch(
+                actions.CR.Nodes.focus(contextPath, fusionPath)
+            );
+        } else {
+            store.dispatch(
+                actions.CR.Nodes.unFocus()
+            );
+        }
+    });
+
+    findAllNodesInGuestFrame().reduce((nodesByPosition, node) => {
+        const {top} = node.getBoundingClientRect();
+
+        if (!nodesByPosition[Math.floor(top / guestFrameWindow.innerHeight)]) {
+            nodesByPosition[Math.floor(top / guestFrameWindow.innerHeight)] = [];
+        }
+
+        nodesByPosition[Math.floor(top / guestFrameWindow.innerHeight)].push(node);
+
+        return nodesByPosition;
+    }, []).forEach((nodeGroup, position) => {
+        const handle = () => {
+            if (Math.floor(guestFrameWindow.scrollY / guestFrameWindow.innerHeight) + 1 >= position) {
+                nodeGroup.forEach(
+                    initializeContentDomNode({
+                        store,
+                        globalRegistry,
+                        nodeTypesRegistry,
+                        inlineEditorRegistry,
+                        nodes
+                    })
+                );
+                guestFrameWindow.removeEventListener('scroll', handle);
+            }
+        };
+
+        guestFrameWindow.addEventListener('scroll', handle);
```

Comment:
I think we should debounce the "handle", so that it is called a little more lazy than on "every" scroll.